src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = structuredClone(state);
+
+  for (const key of actions) {
+    switch (key.type) {
+      case `addProperties`: {
+        Object.assign(stateCopy, key.extraData);
+
+        const newObj = structuredClone(stateCopy);
+
+        stateHistory.push(newObj);
+        break;
+      }
+
+      case `removeProperties`: {
+        for (const propertie of key.keysToRemove) {
```

Comment:
There is a typo in the variable name `propertie`. It should be `property` to correctly iterate over the keys to remove.